### ISSUE TITLE
2021-03-24 nightly compiler crashes when building proc-macro-error

### ISSUE THREAD

**Issue by Krysme** [rust-lang/rust#84026]
rustc version 07e0e2ec2 

How it is reproduced:

OS: Ubuntu 18.04.5 LTS (it can also be reproduced on Arch Linux)
export RUSTFLAGS='-g -C target-cpu=native -Zsanitizer=address'

compile `proc-macro-error` (git version: `a3bad9a2507f0c3b318111297ef6654acca97c37`)

and then I get this error message:

```
thread 'rustc' panicked at 'Failed to get crate data for crate21', compiler/rustc_metadata/src/creader.rs:136:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.53.0-nightly (07e0e2ec2 2021-03-24) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z sanitizer=address -C embed-bitcode=no -C debuginfo=2 -C incremental -C target-cpu=native --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `proc-macro-error`

```

**Comment by jyn514** [comment]
@krysme do you still get the same ICE with `CARGO_INCREMENTAL=0`?

**Comment by Krysme** [comment]
> @Krysme do you still get the same ICE with `CARGO_INCREMENTAL=0`?

Yes, with CARGO_INCREMENTAL=0 exported before `cargo build` the compiler still crashes.

**Comment by est31** [comment]
Sounds like a dupe of #85386, which itself was closed in favour of #56935.

**Comment by Enselic** [comment]
Triage: No protests, so let's close as duplicate.
